crates/iota/src/iota_commands.rs

```diff
@@ -706,6 +706,14 @@ async fn start(
         let network_config_path = config_path.join(IOTA_NETWORK_CONFIG);
         // Auto genesis if no configuration exists in the configuration directory.
         if !network_config_path.exists() {
+            if !config_path.exists() {
+                fs::create_dir(&config_path).map_err(|err| {
+                    anyhow!(err).context(format!(
+                        "Cannot create network config dir {}",
+                        config_path.display()
+                    ))
+                })?;
+            }
```

Comment:
Isn't this covered by `prompt_if_no_config`?

Reply:
No, this here is about the ` --network.config` parameter, so for example if you run `iota start --network.config persisted-localnet` and the folder `persisted-localnet` doesn't exist yet, it will create it